Sources/LeafKit/LeafRenderer.swift

```diff
@@ -21,6 +21,21 @@ final class Cache: LeafCache {
     }
 }
 
+protocol CustomTagProtocol {
+    func render(params: [ProcessedParameter],
+                   body: [Syntax]?,
+                   context: [String: TemplateData]) throws -> TemplateData
+}
+
+struct Lowercased: CustomTagProtocol {
+    func render(params: [ProcessedParameter], body: [Syntax]?, context: [String : TemplateData]) throws -> TemplateData {
```

Comment:
What's the difference between `ProcessedParameter` and `TemplateData`? Could we resolve this to `TemplateData` before passing it to the tag?

Reply:
a processed parameter is something that's been evaluated, so it can be things like an expression, which can't be properly resolved as templatedata here, maybe another name would be more clear about where that fits into the parse/eval process